perun-services/slave/process-k5login_root.sh

```diff
@@ -14,12 +14,20 @@ function process {
 
 	create_lock
 
+	if [ ! -f ${DST_FILE} ]; then
+		# Destination file doesn't exist
+		chown root.root $FROM_PERUN
+		chmod 0644 $FROM_PERUN
+
+		if [`stat -c "%a" ${FROM_PERUN}` -eq 0644 -a `stat -c "%U" ${FROM_PERUN}` == "root" -a `stat -c "%G" ${FROM_PERUN}` == "root" ]; then
```

Comment:
In general it's sufficient to check return codes of commands (in this case chown and chmod) to tell if the action was performed successfully. You don't need to check the result.
To check the return code you can use catch_error function.